src/utils.ts

```diff
@@ -26,3 +32,91 @@ export function toWei(
 export function fromWei(amount: BigNumber, decimals: number): BigNumber {
   return amount.dividedBy(TEN.pow(decimals));
 }
+
+export type SummaryEntry = {
+  tokenAddress: string;
+  amount: BigNumber;
+  decimals: number;
+};
+
+export const transfersToSummary = (transfers: Payment[]) => {
+  return transfers.reduce((previousValue, currentValue): Map<
+    string,
+    SummaryEntry
+  > => {
+    let tokenSummary = previousValue.get(currentValue.tokenAddress);
+    if (typeof tokenSummary === "undefined") {
+      tokenSummary = {
+        tokenAddress: currentValue.tokenAddress,
+        amount: new BigNumber(0),
+        decimals: currentValue.decimals,
+      };
+      previousValue.set(currentValue.tokenAddress, tokenSummary);
+    }
+    tokenSummary.amount = tokenSummary.amount.plus(currentValue.amount);
+
+    return previousValue;
+  }, new Map<string, SummaryEntry>());
+};
+
+export type InsufficientBalanceInfo = {
+  token: string;
+  transferAmount: string;
+};
+
+export const checkAllBalances = async (
+  summary: Map<string, SummaryEntry>,
+  web3Provider: ethers.providers.Web3Provider,
+  safe: SafeInfo,
+  tokenList: TokenMap
+): Promise<InsufficientBalanceInfo[]> => {
+  const insufficientTokens: InsufficientBalanceInfo[] = [];
+  for (const summaryEntry of summary.values()) {
+    const tokenAddress = summaryEntry.tokenAddress;
+    const tokenAmount = summaryEntry.amount;
```

Comment:
Can we destructure here inside the { tokenAddress, amount } of `summary.values()`?